lib/babe/babe.go

```diff
@@ -415,79 +376,33 @@ func (b *Service) invokeBlockAuthoring() error {
 			case <-b.pause:
 				cleanup()
 				return nil
-			case <-slotDone[i]:
-				slotNum := startSlot + uint64(i)
-				err = b.handleSlot(epoch, slotNum)
-				if err == ErrNotAuthorized {
-					logger.Debugf(
-						"not authorized to produce a block in slot %d, at epoch %d with %d slots in this epoch",
-						slotNum, epoch, slotNum-epochStartSlot)
-					continue
-				} else if err != nil {
-					logger.Warnf("failed to handle slot %d: %s", slotNum, err)
-					continue
-				}
 			case <-epochTimer.C:
-				done = true
+				// stop current epoch handler
+				cancel()
```

Comment:
when the `epochTimer.C` gets triggered should we just let the code flow until the next return statement to call the `defer cancel()` function?
```suggestion
```

Reply:
no, because we want to cancel the current epoch, then do the `incrementEpoch` call